go/platform/apple.bzl

```diff
@@ -0,0 +1,58 @@
+APPLE_GOOS = {
+    "@io_bazel_rules_go//go/platform:ios_i386": "darwin",
+    "@io_bazel_rules_go//go/platform:ios_x86_64": "darwin",
+    "@io_bazel_rules_go//go/platform:ios_arm64": "darwin",
+    "@io_bazel_rules_go//go/platform:ios_armv7": "darwin",
+}
+
+APPLE_GOARCH = {
+    "@io_bazel_rules_go//go/platform:ios_i386": "386",
+    "@io_bazel_rules_go//go/platform:ios_x86_64": "amd64",
+    "@io_bazel_rules_go//go/platform:ios_arm64": "arm64",
+    "@io_bazel_rules_go//go/platform:ios_armv7": "arm",
+}
+
+PLATFORMS = {
+    'armv7-apple-ios': apple_common.platform.ios_device,
+    'armv7-apple-tvos': apple_common.platform.tvos_device,
+
+    'armv7k-apple-watchos': apple_common.platform.watchos_device,
+
+    'arm64-apple-ios': apple_common.platform.ios_device,
+    'arm64-apple-tvos': apple_common.platform.tvos_device,
+
+    'i386-apple-ios': apple_common.platform.ios_simulator,
+    'i386-apple-macosx': apple_common.platform.macos,
+    'i386-apple-tvos': apple_common.platform.tvos_simulator,
+    'i386-apple-watchos': apple_common.platform.watchos_simulator,
+
+    'x86_64-apple-ios': apple_common.platform.ios_simulator,
+    'x86_64-apple-macosx': apple_common.platform.macos,
+    'x86_64-apple-tvos': apple_common.platform.ios_simulator,
+    'x86_64-apple-watchos': apple_common.platform.watchos_simulator,
+}
+
+def apple_declare_config_settings():
+    for cpu in ["i386", "x86_64", "armv7", "arm64"]:
```

Comment:
Style nit: 2-space indent for blocks (function bodies, `if`, `for`, etc). The 4-space indent for list / dict literals and function calls is correct though.

Reply:
You know I contemplated doing a PR fixing it to 4 for all files :)
Where does this rule come from? Having mixed indentation kind of kills my editor :(

Reply:
Your wish will soon be granted: #1458. The new buildifier formats .bzl files and it does a 4-space indent everywhere, so we'll switch to that in the near future.